helm/lib/dependabot/helm/update_checker/latest_version_resolver.rb

```diff
@@ -0,0 +1,192 @@
+        ).void
+      end
+      def initialize(dependency:, credentials:, cooldown_options:)
+        @dependency = dependency
+        @credentials = credentials
+        @cooldown_options = cooldown_options
+      end
+
+      sig { returns(Dependabot::Dependency) }
+      attr_reader :dependency
+
+      # To filter versions in cooldown period based on version tags from registry call
+      sig do
+        params(tags: T::Array[String], tags_with_release_date: T::Array[GitTagWithDetail])
+          .returns(T::Array[String])
+      end
+      def filter_versions_in_cooldown_period_using_oci(tags, tags_with_release_date)
+        select_tags_which_in_cooldown_using_oci(tags_with_release_date)&.each do |tag_name|
+          # Iterate through versions and filter out those matching the tag_name
+          tags.reject! do |version|
+            version == tag_name
+          end
+        end
+        tags
+      rescue StandardError => e
+        Dependabot.logger.error("Error filter_versions_in_cooldown_period_for_oci:: #{e.message}")
+        tags
+      end
+
+      sig do
+        params(
+          versions: T::Array[T::Hash[String, T.untyped]],
+          repo_name: T.nilable(String),
+          chart_name: T.nilable(String)
+        )
+          .returns(T::Array[T::Hash[String, T.untyped]])
+      end
+      def fetch_tag_and_release_date_helm_chart(versions, repo_name, chart_name)
+        Dependabot.logger.info("Filtering versions in cooldown period from chart: #{repo_name}")
+        # Using index URL to fetch tags in cooldown period"
+        tags = select_tags_which_in_cooldown_from_chart_index("", T.must(chart_name))
+        # If no tags in result then check from github api.
+        tags = select_tags_which_in_cooldown_from_chart(T.must(repo_name)) if tags.nil? || tags.empty?
+
+        return versions if tags.nil? || tags.empty?
+
+        versions.reject! do |release|
+          tags.any?(release["version"])
+        end
+        versions
+      rescue StandardError => e
+        Dependabot.logger.error("Error fetch_tag_and_release_date_helm_chart(versions): #{e.message}")
+        versions
+      end
+
+      sig { params(repo_name: String).returns(T.nilable(T::Array[String])) }
+      def select_tags_which_in_cooldown_from_chart(repo_name)
+        version_tags_in_cooldown_from_chart = T.let([], T::Array[String])
+
+        begin
+          package_details_fetcher.fetch_tag_and_release_date_from_chart(repo_name).each do |git_tag_with_detail|
+            if check_if_version_in_cooldown_period?(T.must(git_tag_with_detail.release_date))
+              version_tags_in_cooldown_from_chart << git_tag_with_detail.tag
+            end
+          end
+          version_tags_in_cooldown_from_chart
+        rescue StandardError => e
+          Dependabot.logger.error("Error checking if version is in cooldown: #{e.message}")
+          version_tags_in_cooldown_from_chart
```

Comment:
Again, we can use an `ensure` block here:


```suggestion
        rescue StandardError => e
          Dependabot.logger.error("Error checking if version is in cooldown: #{e.message}")
        ensure
          version_tags_in_cooldown_from_chart
```